pinot-spi/src/main/java/org/apache/pinot/spi/config/table/SegmentsValidationAndRetentionConfig.java

```diff
@@ -52,6 +52,10 @@ public class SegmentsValidationAndRetentionConfig extends BaseJsonConfig {
   // Number of replicas per partition of low-level consumers. This config is used for realtime tables only.
   private String _replicasPerPartition;
 
+  // True if and only if the segment upload to deep store is disabled. This is usefully for Pinot cluster which does not
+  // have any deep store.
+  private boolean _segmentUploadToDeepStoreDisabled;
```

Comment:
we shall also disallow rebalance with downtime when this is disabled, right?

Reply:
Yes. If there is no deep store copy, the rebalance with downtime will result in segment data loss. I will add a note about this mode of ingestion with its limitation. We can also file a PR to detect the missing segments during rebalance.